pkgs/top-level/all-packages.nix

```diff
@@ -9761,6 +9761,10 @@ in
 
   leaps = callPackage ../development/tools/leaps { };
 
+  ### DEVELOPMENT / JAVA MODULES
+
+  javaPackages = recurseIntoAttrs (callPackage ./java-packages.nix { });
```

Comment:
I actually think this should be `callPackages` rather than `callPackage`. The difference is that the former recurses (one level) into the inner attrset and makes the individual derivations there overridable. I think you'll find that in the current setup, `override[Derivation]` won't work on anything inside there.